repo/maintenance/maintenance_run.go

```diff
@@ -337,9 +374,40 @@ func runTaskCleanupEpochManager(ctx context.Context, runParams RunParameters, s
 		return nil
 	}
 
-	return ReportRun(ctx, runParams.rep, TaskCleanupEpochManager, s, func() error {
+	if err := runTaskEpochAdvance(ctx, em, runParams, s); err != nil {
+		return err
+	}
+
+	// compact a single epoch
+	if err := ReportRun(ctx, runParams.rep, TaskEpochCompactSingle, s, func() error {
+		log(ctx).Infof("Compacting an eligible uncompacted epoch...")
+		return errors.Wrap(em.MaybeCompactSingleEpoch(ctx), "error compacting single epoch")
```

Comment:
Can you explain a bit about the ordering in full vs quick maintenance? Here we `runTaskEpochAdvance` then `MaybeCompactSingleEpoch`. In quick we do it in reverse order. Is there any significance, does the ordering matter at all? Will we be writing compacted blobs into old/new epochs depending on quick vs full?

Reply:
There is no constraint in the ordering. We should be consistent though.
It may make sense to advance first.